From 14287d1e90e9a68c3dc276a4b610a45bdb7911c9 Mon Sep 17 00:00:00 2001
From: Omkar Anand Kulkarni <omkar.kulkarni@arm.com>
Date: Fri, 12 Apr 2024 07:56:08 +0530
Subject: [PATCH] feat(neoverse): introduce build option for platform error
 injection

Platforms that support ras error handling either using firmware-first or
kernel-first error handling approach require error injection software to
validate the error handling functionality. Define a new build option
NRD_ERR_INJ_SUPPORT to enable or disable platform error injection
feature.

Signed-off-by: Omkar Anand Kulkarni <omkar.kulkarni@arm.com>
Change-Id: Ia4dfb9e5ed9db31395efc96d16872e11d21135f1

Signed-off-by: Jon Mason <jon.mason@arm.com>
Upstream-Status: Pending
---
 docs/plat/arm/arm-build-options.rst             | 6 ++++++
 plat/arm/board/neoverse_rd/common/nrd-common.mk | 4 ++++
 2 files changed, 10 insertions(+)

diff --git a/docs/plat/arm/arm-build-options.rst b/docs/plat/arm/arm-build-options.rst
index 3fad566940c5..c75890099cf1 100644
--- a/docs/plat/arm/arm-build-options.rst
+++ b/docs/plat/arm/arm-build-options.rst
@@ -157,6 +157,12 @@ Arm Neoverse RD Platform Build Options
    valid value greater than 1, the platform code performs required configuration
    to support multi-chip operation.
 
+-  ``NRD_ERR_INJ_SUPPORT``: Configures error injection support on SGI/RD
+   platform. Error injection software implementation is used to validate the
+   error handling software that is implemented by the platform or the OS.
+   Define a build option to selectively enable or disable the error injection
+   platform support for a given platform.
+
 - ``NRD_PLATFORM_VARIANT``: Selects the variant of a Neoverse RD platform. A
   particular Neoverse RD platform may have multiple variants which may differ in
   core count, cluster count or other peripherals. This build option is used to
diff --git a/plat/arm/board/neoverse_rd/common/nrd-common.mk b/plat/arm/board/neoverse_rd/common/nrd-common.mk
index eb17f53a5a0c..d8621f74418d 100644
--- a/plat/arm/board/neoverse_rd/common/nrd-common.mk
+++ b/plat/arm/board/neoverse_rd/common/nrd-common.mk
@@ -20,6 +20,8 @@ NRD_CHIP_COUNT		:=	1
 
 NRD_PLATFORM_VARIANT	:=	0
 
+NRD_ERR_INJ_SUPPORT	:=	0
+
 # Do not enable SVE
 ENABLE_SVE_FOR_NS		:=	2
 ENABLE_SVE_FOR_SWD		:=	1
@@ -52,6 +54,8 @@ $(eval $(call add_define,NRD_CHIP_COUNT))
 
 $(eval $(call add_define,NRD_PLATFORM_VARIANT))
 
+$(eval $(call add_define,NRD_ERR_INJ_SUPPORT))
+
 override CSS_LOAD_SCP_IMAGES	:=	0
 override NEED_BL2U		:=	no
 override ARM_PLAT_MT		:=	1
